packages/core/src/services/files/transform.ts

```diff
@@ -0,0 +1,107 @@
+// @ts-expect-error istextorbinary is not typed
+import { isText } from 'istextorbinary'
+import { parseOfficeAsync } from 'officeparser'
+import path from 'path'
+import * as pdfjs from 'pdfjs-dist'
+import type { TextItem } from 'pdfjs-dist/types/src/display/api'
+import {
+  BadRequestError,
+  Result,
+  TypedResult,
+  UnprocessableEntityError,
+} from '../../lib'
+
+import { MAX_UPLOAD_SIZE_IN_MB } from '../../constants'
+
+export async function transformFile(
+  file: File,
+): Promise<TypedResult<string, Error>> {
+  const extension = path.extname(file.name).toLowerCase()
+
+  if (file.size === 0) {
+    return Result.error(new BadRequestError(`File is empty`))
+  }
+
+  if (file.size > MAX_UPLOAD_SIZE_IN_MB) {
+    return Result.error(new BadRequestError(`File too large`))
+  }
+
+  const buffer = Buffer.from(await file.arrayBuffer())
+  let content = ''
+
+  try {
+    switch (extension) {
+      case '.pdf':
+        content = await transformPdfFile(buffer)
+        break
+      case '.docx':
+      case '.xlsx':
+      case '.pptx':
+      case '.odt':
+      case '.ods':
+      case '.odp':
+        content = await parseOfficeAsync(buffer, {
+          outputErrorToConsole: false,
+          newlineDelimiter: '\n',
+          ignoreNotes: false,
+          putNotesAtLast: false,
+        })
+        break
+      default:
+        if (!isText(file.name, buffer)) {
+          return Result.error(
+            new BadRequestError(`Unsupported file type: ${extension}`),
+          )
+        }
+        content = buffer.toString()
```

Comment:
So a `.md` or `.text` will arrive here no?

Reply:
yes, any file that is raw text